src/defs.hpp

```diff
@@ -86,6 +69,45 @@ struct RegionSize { // aggregate and POD type; do NOT reorder member declaration
 // X3DIR z, phi, etc...
 enum CoordinateDirection { NODIR = -1, X0DIR = 0, X1DIR = 1, X2DIR = 2, X3DIR = 3 };
 
+struct RegionSize {
+  RegionSize() = default;
+  RegionSize(std::array<Real, 3> xmin, std::array<Real, 3> xmax, std::array<Real, 3> xrat,
+             std::array<int, 3> nx)
+      : xmin_(xmin), xmax_(xmax), xrat_(xrat),
+        nx_(nx), symmetry_{nx[0] == 1, nx[1] == 1, nx[2] == 1} {}
+  RegionSize(std::array<Real, 3> xmin, std::array<Real, 3> xmax, std::array<Real, 3> xrat,
+             std::array<int, 3> nx, std::array<bool, 3> symmetry)
+      : xmin_(xmin), xmax_(xmax), xrat_(xrat), nx_(nx), symmetry_(symmetry) {}
+
+  std::array<Real, 3> xmin_, xmax_, xrat_; // xrat is ratio of dxf(i)/dxf(i-1)
+  std::array<int, 3> nx_;
+  std::array<bool, 3> symmetry_;
+
+  Real &xmin(CoordinateDirection dir) { return xmin_[dir - 1]; }
+  const Real &xmin(CoordinateDirection dir) const { return xmin_[dir - 1]; }
+
+  Real &xmax(CoordinateDirection dir) { return xmax_[dir - 1]; }
+  const Real &xmax(CoordinateDirection dir) const { return xmax_[dir - 1]; }
+
+  Real &xrat(CoordinateDirection dir) { return xrat_[dir - 1]; }
+  const Real &xrat(CoordinateDirection dir) const { return xrat_[dir - 1]; }
+
+  int &nx(CoordinateDirection dir) { return nx_[dir - 1]; }
+  const int &nx(CoordinateDirection dir) const { return nx_[dir - 1]; }
+
+  bool &symmetry(CoordinateDirection dir) { return symmetry_[dir - 1]; }
+  const bool &symmetry(CoordinateDirection dir) const { return symmetry_[dir - 1]; }
+};
+
+//----------------------------------------------------------------------------------------
+// enums used everywhere
+// (not specifying underlying integral type (C++11) for portability & performance)
+
+// TODO(felker): C++ Core Guidelines Enum.5: Don’t use ALL_CAPS for enumerators
```

Comment:
leftovers from Athena++

Reply:
I think this all goes in one of the subsequent GMG PRs